diff --git a/master/buildbot/process/build.py b/master/buildbot/process/build.py
index 3477444..000999e 100644
--- a/master/buildbot/process/build.py
+++ b/master/buildbot/process/build.py
@@ -270,6 +270,9 @@ class Build(properties.PropertiesMixin):
 
         self.build_status.buildStarted(self)
         self.acquireLocks().addCallback(self._startBuild_2)
+        if (self.slavebuilder.slave.properties.getProperty('oneshot', False)):
+            log.msg("oneshot slave started, requesting graceful shutdown.")
+            self.slavebuilder.slave.slave_status.setGraceful(True)
         return d
 
     def acquireLocks(self, res=None):
