diff --git a/master/setup.py b/master/setup.py
index 89bda2f..18a06d2 100755
--- a/master/setup.py
+++ b/master/setup.py
@@ -189,7 +189,7 @@ else:
     setup_args['install_requires'] = [
         'twisted >= 9.0.0',
         'Jinja2 >= 2.1',
-        'sqlalchemy >= 0.6, < 0.8',
+        'sqlalchemy >= 0.6, <= 0.7.9',
         # buildbot depends on sqlalchemy internals, and these are the tested
         # versions.
         'sqlalchemy-migrate ==0.6.1, ==0.7.0, ==0.7.1, ==0.7.2',
